assets/src/dashboard/app/api/useTemplateApi.js

```diff
@@ -0,0 +1,136 @@
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useState, useMemo, useCallback } from 'react';
+import moment from 'moment';
+
+/**
+ * Internal dependencies
+ */
+import getAllTemplates from '../../templates';
+import { APP_ROUTES } from '../../constants';
+
+function reshapeTemplateObject(isLocal) {
+  return ({ id, title, tags, colors, createdBy, description, pages }) => ({
+    isLocal,
+    id,
+    title,
+    createdBy,
+    description,
+    status: 'template',
+    modified: moment('2020-04-07'),
+    metadata: [...tags, ...colors].map((value) =>
+      typeof value === 'object' ? { ...value } : { label: value }
+    ),
+    pages,
+    centerTargetAction: `#${APP_ROUTES.TEMPLATE_DETAIL}?id=${id}&isLocal=${isLocal}`,
+    bottomTargetAction: () => {},
+  });
+}
+
+// TODO: Remove this eslint rule once endpoints are working
+/* eslint-disable no-unused-vars */
+const useTemplateApi = (dataAdapter, config) => {
+  const [templates, setTemplates] = useState([]);
+  const { pluginDir } = config;
+
+  const createStoryFromTemplatePages = useCallback((pages) => {
+    return Promise.resolve({ success: true, storyId: -1 });
+  }, []);
+
+  const fetchSavedTemplates = useCallback((filters) => {
+    // Saved Templates = Bookmarked Templates + My Templates
+    setTemplates([]);
+    return Promise.resolve([]);
+  }, []);
+
+  const fetchBookmarkedTemplates = useCallback((filters) => {
+    setTemplates([]);
+    return Promise.resolve([]);
+  }, []);
+
+  const fetchMyTemplates = useCallback((filters) => {
+    setTemplates([]);
+    return Promise.resolve([]);
+  }, []);
+
+  const fetchMyTemplateById = useCallback((templateId) => {
+    return Promise.resolve({});
+  }, []);
+
+  const fetchExternalTemplates = useCallback(
+    (filters) => {
+      const reshapedTemplates = getAllTemplates({ pluginDir }).map(
+        reshapeTemplateObject(false)
+      );
+      setTemplates(reshapedTemplates);
+      return Promise.resolve(reshapedTemplates);
+    },
+    [pluginDir]
+  );
+
+  const fetchExternalTemplateById = useCallback(
+    async (templateId) => {
+      const fetchedTemplates = await fetchExternalTemplates();
+
+      return Promise.resolve(
+        fetchedTemplates.find((template) => template.id === templateId)
+      );
+    },
+    [fetchExternalTemplates]
+  );
+
+  const bookmarkTemplate = useCallback((templateId, shouldBookmark) => {
+    if (shouldBookmark) {
+      // api call to bookmark template
+      return Promise.resolve({ success: true });
+    } else {
+      // api call to remove bookmark from template
+      return Promise.resolve({ success: true });
+    }
+  }, []);
+
+  const api = useMemo(
+    () => ({
+      bookmarkTemplate,
+      createStoryFromTemplatePages,
+      fetchBookmarkedTemplates,
+      fetchSavedTemplates,
+      fetchMyTemplates,
+      fetchMyTemplateById,
+      fetchExternalTemplates,
+      fetchExternalTemplateById,
+    }),
```

Comment:
These are just stubbed out endpoints so we can continue to hook up the template UI.